wordpress-mobile/WordPress-Android
# Stats revamp v2 detail screens date pickers

This PR adds date selector on Insights detail screens

NOTE:
- This PR only adds a date selector feature.  Cards on each detail screen may not be working yet with date selection
- _Each individual detail screen needs a separate PR to make sure the date selection actually reflects in the cards on the respective screen._

Fixes an outstanding item from 16684 


|Detail screen with date selector|
|---|
|<img width=320 src="https://user-images.githubusercontent.com/990349/171578960-4edc2f08-8b68-46a7-8b35-0bdefa10f011.png" />|


To test:

Setup:
1. Go to App Settings (Tap on Avatar at the top right hand corner on My Site -> Find App Settings)
2. Select Debug Settings
3. Find StatsRevampV2FeatureConfig under Features in development enable it and restart app
4. Launch / Re-Launch app
5. Go to Stats either using quick links or menu
6. Switch to Insights tab if necessary
7. Make sure Views & Visitors, Total Likes, Total Comments and Total Followers cards are visible on Insights tab. If those cards are not present then add them through stats management using either [+ add new stats card ] at the bottom or clicking ⚙️ icon at the top right hand corner.

Test:
- Click on View More on Views & Visitors, Total Likes, Total Comments and Total Followers cards.
- Ensure it takes to respective detail screen
- Ensure a date selector is visible at the top 
- Make sure the title represents the respective card.  For e.g. Total Likes


## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing all the paths

3. What automated tests I added (or what prevented me from doing so)
updated existing unit tests

PR submission checklist:
